src/pharmpy/tools/amd/run.py

```diff
@@ -181,16 +212,26 @@ def run_amd(
     run_subfuncs = {}
     for section in order:
         if section == 'structural':
-            func = _subfunc_modelsearch(search_space=modelsearch_features, path=db.path)
-            run_subfuncs['modelsearch'] = func
+            if modeltype == 'pkpd':
+                func = _subfunc_structsearch(
+                    route=administration,
+                    modeltype=modeltype,
+                    emax_init=emax_init,
+                    ec50_init=ec50_init,
+                    path=db.path,
+                )
+                run_subfuncs['structsearch'] = func
+            else:
+                func = _subfunc_modelsearch(search_space=modelsearch_features, path=db.path)
+                run_subfuncs['modelsearch'] = func
         elif section == 'iivsearch':
             func = _subfunc_iiv(path=db.path)
             run_subfuncs['iivsearch'] = func
         elif section == 'iovsearch':
             func = _subfunc_iov(amd_start_model=model, occasion=occasion, path=db.path)
             run_subfuncs['iovsearch'] = func
         elif section == 'residual':
-            func = _subfunc_ruvsearch(path=db.path)
+            func = _subfunc_ruvsearch(dv=dv, path=db.path)
```

Comment:
What will dv be for pk-models?

Reply:
The default value for dv is None. The default value for dv is None in the error models as well.

Reply:
The default value will never be used since it is the `dv` argument is set to `dv` that hasn't been assigned any value. Or was it assigned somewhere that I missed?

Reply:
The default dv is set in the input arguments of the AMD tool in l. 52.

Reply:
Ok, i missed that. It feels strange to have the dv option for the amd tool. What is it supposed to do?

Reply:
It is needed for the ruvsearch tool. But since I am setting dv = 2 for pkpd models in l. 143 I guess it is not needed.